File: QuanLyPhongKham.Server/src/QuanLyPhongKham.Models/Entities/DANHMUCTHUOC.cs
using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.Models.Entities
{
    public class DANHMUCTHUOC : BaseEntity
    {
        [Key]
        public Guid MaThuoc { get; set; }
        public string TenThuoc { get; set; }
        public decimal DonGia { get; set; }
        public string? ChongChiDinh { get; set; }

        public ICollection<CHITIETDONTHUOC>? ChiTietDonThuocs { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/models/DanhMucThuoc.js
/**
 * @typedef {Object} DanhMucThuoc
 * @property {string} maThuoc
 * @property {string} tenThuoc
 * @property {number} donGia
 * @property {string} chongChiDinh
 */

export const createDanhMucThuoc = (
  maThuoc = "",
  tenThuoc = "",
  donGia = 0,
  chongChiDinh = "",
) => ({
  maThuoc,
  tenThuoc,
  donGia,
  chongChiDinh,
});

export const normalizeDanhMucThuoc = (item) => ({
  maThuoc: item?.maThuoc ?? item?.MaThuoc ?? "",
  tenThuoc: item?.tenThuoc ?? item?.TenThuoc ?? "",
  donGia: item?.donGia ?? item?.DonGia ?? 0,
  chongChiDinh: item?.chongChiDinh ?? item?.ChongChiDinh ?? "",
});
